modoboa/admin/management/commands/subcommands/_mx.py

```diff
@@ -0,0 +1,201 @@
+                if address is not None:
+                    record = models.MXRecord.objects.create(
+                        domain=domain,
+                        name=str(answer.exchange).strip('.'),
+                        address=address,
+                        updated=now)
+                    yield record
+
+    def query_dnsbl(self, mx_list, provider):
+        """Check given IP against given provider."""
+        results = {}
+        for mx in mx_list:
+            reverse = ".".join(reversed(mx.address.split(".")))
+            pattern = "{}.{}".format(reverse, provider)
+            try:
+                results[mx] = socket.gethostbyname(pattern)
+            except socket.gaierror:
+                results[mx] = False
+        return provider, results
+
+    def store_dnsbl_result(self, domain, provider, results):
+        """Store provider results for domain."""
+        alerts = {}
+        to_create = []
+        for mx in results.keys():
+            result = "" if not results[mx] else results[mx]
+            dnsbl_result = models.DNSBLResult.objects.filter(
+                domain=domain, provider=provider, mx=mx).first()
+            if dnsbl_result is None:
+                to_create.append(
+                    models.DNSBLResult(
+                        domain=domain, provider=provider, mx=mx,
+                        status=result)
+                )
+            else:
+                if not dnsbl_result.status and result:
+                    if domain not in alerts:
+                        alerts[domain] = []
+                    alerts[domain].append((provider, mx))
+                dnsbl_result.status = result
+                dnsbl_result.save()
+        models.DNSBLResult.objects.bulk_create(to_create)
+        if not alerts:
+            return
+        content = render_to_string(
+            "admin/notifications/domain_in_dnsbl.html", {
+                "domain": domain, "alerts": alerts
+            })
+        subject = _("[modoboa] DNSBL issue(s) for domain {}").format(
+            domain.name)
+        for admin in domain.admins:
+            if not admin.email:
+                continue
+            status, msg = email_utils.sendmail_simple(
+                self.sender, admin.email,
+                subject=subject, content=content)
+            if not status:
+                print(msg)
+
+    def check_valid_mx(self, domain, mx_list, valid_mx=None, **options):
+        alerts = []
+        check = False
+        mx_names = [mx.name for mx in mx_list]
+        mxs = mx_names + [mx.address for mx in mx_list]
+        if valid_mx:
+            check = True in [mx in mxs for mx in valid_mx]
+        if not mxs:
+            alerts.append(_('Domain {} as no MX record').format(domain))
+        elif valid_mx and check is False:
+            args = (domain, ', '.join(mx_names))
+            alerts.append(
+                _('Domain {0} got an invalid MX record. Got {1}').format(*args)
```

Comment:
Could we avoid mixing '' and "" ? Modoboa is not perfect yet but I'm working on it and I prefer double quotes.